eth/config.go

```diff
@@ -51,7 +51,7 @@ var DefaultLightGPOConfig = gasprice.Config{
 var DefaultConfig = Config{
 	SyncMode: downloader.FastSync,
 	Ethash: ethash.Config{
-		CacheDir:         "ethash",
+		CacheDir:         "etchash",
```

Comment:
This config change causes the Ethash/Etchash default directory to _always_ be `~/.etchash` and `<datadir>/etchash`. Even for ETH chains, et al. This would mean that clients using core-geth on ETH would not only need to regenerate their DAG needlessly, but it would also break their filesystem API, since they'll expect that dir to be in `~/.ethash`.

We should only use a new directory default (and DAG!) for the configurations for which it's necessary.